### PR TITLE
KAFKA-16806: Explicitly declare JUnit dependencies for all test modules

### PR DESCRIPTION
**Pull request by frankvicky**
The automatic loading of test framework implementation dependencies has been deprecated and is scheduled for removal in Gradle 9.0, so we need to declare the desired test framework directly on the test suite or explicitly declare the test framework implementation dependencies on the test's runtime classpath.  

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -53,6 +54,14 @@ plugins {
   id 'com.diffplug.spotless' version "6.14.0" apply false
 }
 
+testing {
```
Comment (chia7712) [inline build.gradle]: Could you please run it with `unitTest` and `integrationTest`? Also, please make sure the junit 4 style tests still works?

Comment (frankvicky) [reply build.gradle]: I will take a look

**Review by chia7712 — APPROVED** [review]
LGTM

**Comment by frankvicky** [comment]
Since we have remove all dependencies of JUnit 4, i think the PR is ready for review.

File: build.gradle [hunk 2]
```diff
@@ -35,6 +35,7 @@ plugins {
   id 'idea'
   id 'jacoco'
   id 'java-library'
+  id 'jvm-test-suite'
```
Comment (chia7712) [inline build.gradle]: We have introduced dependency of junit platform (https://github.com/apache/kafka/blob/trunk/gradle/dependencies.gradle#L204), so maybe we can manually declare dependencies?

File: gradle/dependencies.gradle [hunk 3]
```diff
@@ -202,6 +202,7 @@ libs += [
   junitJupiter: "org.junit.jupiter:junit-jupiter:$versions.junit",
   junitJupiterApi: "org.junit.jupiter:junit-jupiter-api:$versions.junit",
   junitPlatformSuiteEngine: "org.junit.platform:junit-platform-suite-engine:$versions.junitPlatformSuiteEngine",
+  junitPlatformLanucher: "org.junit.platform:junit-platform-launcher",
```
Comment (chia7712) [inline gradle/dependencies.gradle]: please include the version declaration

**Comment by frankvicky** [comment]
Hi @chia7712 
I have specify the version of `junit-platform-launcher`, PTAL

File: gradle/dependencies.gradle [hunk 4]
```diff
@@ -157,7 +157,8 @@ versions += [
   zinc: "1.9.2",
   zookeeper: "3.8.4",
   zstd: "1.5.6-3",
-  junitPlatformSuiteEngine: "1.10.2"
+  junitPlatformSuiteEngine: "1.10.2",
```
Comment (chia7712) [inline gradle/dependencies.gradle]: Could you please merge those two version variables? Both of them are about junit platform, right?

Comment (frankvicky) [reply gradle/dependencies.gradle]: Yep, I forget they are all under junitPlatform and share same version. I will combine them.

**Comment by frankvicky** [comment]
Hi @chia7712 
I have unify the version reference of `junitPlatform` related dependencies, PTAL

**Comment by chia7712** [comment]
@gharris1727 which solution you prefer to apply to kafka? We have introduced junit platform-related deps to kafka ( for suite test in streams). Hence, declaring dependencies is better to me since they can be under the same version control.

**Comment by gharris1727** [comment]
@chia7712 I don't understand the question. What is the alternative to this PR? Does this solve the build warning?

**Comment by chia7712** [comment]
> What is the alternative to this PR?

another solution recommended by gradle is using `jvm-test-suite`. for example:
```
testing {
    suites { 
        test { 
            useJUnitJupiter() 
        }
    }
}
```
However, it is incubating so I prefer to declare the junit platform explicitly.

> Does this solve the build warning?

yep. see following console output

### before
```
> Task :connect:mirror-client:test
The automatic loading of test framework implementation dependencies has been deprecated. This is scheduled to be removed in Gradle 9.0. Declare the desired test framework directly on the test suite or explicitly declare the test framework implementation dependencies on the test's runtime classpath. Consult the upgrading guide for further information: https://docs.gradle.org/8.8/userguide/upgrading_version_8.html#test_framework_implementation_dependencies
```

### after
```
> Task :connect:mirror-client:test

Gradle Test Run :connect:mirror-client:test > Gradle Test Executor 146 > SourceAndTargetTest > testEquals() PASSED
```

**Review by gharris1727 — APPROVED** [review]
> However, it is incubating so I prefer to declare the junit platform explicitly.

I agree.

> yep. see following console output

Very nice, :+1:

**Comment by gharris1727** [comment]
Thanks @frankvicky for the fix!

**Pull request merged**